https://bugzilla.netfilter.org/show_bug.cgi?id=1637

From a450f6374dc1b0296578599adf77f9ac025fab85 Mon Sep 17 00:00:00 2001
From: Sam James <sam@gentoo.org>
Date: Thu, 24 Nov 2022 07:51:23 +0000
Subject: [PATCH 1/3] Makefile.am: don't suppress various warnings

These will become fatal with Clang 16 and GCC 14 anyway, but let's
address the real problem (followup commit).

We do have to keep one wrt yyerror() & const char* though, but
the issue is contained to the code Bison generates.

Bug: https://bugzilla.netfilter.org/show_bug.cgi?id=1637
Signed-off-by: Sam James <sam@gentoo.org>
--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -60,8 +60,8 @@ if HAVE_SYSTEMD
 conntrackd_SOURCES += systemd.c
 endif
 
-# yacc and lex generate dirty code
-read_config_yy.o read_config_lex.o: AM_CFLAGS += -Wno-missing-prototypes -Wno-missing-declarations -Wno-implicit-function-declaration -Wno-nested-externs -Wno-undef -Wno-redundant-decls -Wno-sign-compare
+# yacc and lex generate dirty code (issue is in yyerror() wrt const char* vs. char*)
+read_config_yy.o read_config_lex.o: AM_CFLAGS += -Wno-incompatible-pointer-types -Wno-discarded-qualifiers
 
 conntrackd_LDADD = ${LIBMNL_LIBS} ${LIBNETFILTER_CONNTRACK_LIBS} \
 		   ${libdl_LIBS} ${LIBNFNETLINK_LIBS}
